File: pset1/mario/more/mario.c
#include <stdbool.h>
#include <stdio.h>
#include <cs50.h>

int prompt_height(int max);
void make_pyramid(int height);

// main
int main(void)
{

    // prompt user for pyramid height
    int height = prompt_height(8);

    // call function to generate pyramid
    make_pyramid(height);
}


// prompt user to input a height for pyramid
int prompt_height(int max)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height > max || height < 1);

    return height;
}


// generate pyramid based on input height
void make_pyramid(int height)
{
    int nr_spaces = height - 1;
    int nr_hashes = 0;
    // loop through matrix
    for (int row = 0; row < height; row++)
    {
        for (int col = 0; col < height; col++)
        {
            if (col >= nr_spaces)
            {
                // print hashes
                printf("#");
            }
            else
            {
                // print spaces
                printf(" ");
            }
        }
        // print gap
        printf("  ");
        for (int col = 0; col < height; col++)
        {
            if (col <= nr_hashes)
            {
                printf("#");
            }
        }
        nr_spaces--;
        nr_hashes++;
        printf("\n");
    }
}

File: problemSets/pset6/mario/more/mario.py
from cs50 import get_int

# main function


def main():
    # create pyramid object and prompt height
    pyramid = Pyramid()
    # build pyramid
    pyramid.build_pyramid()


class Pyramid:
    """This Pyramid will take in a max height and a pyramid height
    which will then check to see if pyramid height falls between zero and max height
    lastly it will print out a pyramid of given height"""

    # __init__
    def __init__(self, max_height: int = 8):
        self.height: int = 0
        self.max_height = max_height
        # if the height does not satisfy requirements: REPROMPT
        while self.height > self.max_height or self.height < 1:
            self.height = get_int("Height: ")

    def build_pyramid(self):
        # grid comprised of spaces and hashes
        # as iterate through rows, nrspaces decr & nrhashes incr
        nr_spaces: int = self.height - 1
        nr_hashes: int = 0
        # for each loc determine is print # or ' '
        for row in range(self.height):
            for col in range(self.height):
                if col >= nr_spaces:
                    print("#", end='')
                else:
                    print(" ", end='')
            print("  ", end='')
            for col in range(self.height):
                if col <= nr_hashes:
                    print("#", end='')
            nr_spaces -= 1
            nr_hashes += 1
            print("")


# call main function
if __name__ == '__main__':
    main()
